backend/graminsta/post/models.py

```diff
@@ -0,0 +1,29 @@
+"""
+Here defines the database tables for post app
+"""
+
+from django.db import models
+from django.contrib.auth.models import User
+
+
+class Post(models.Model):
+    """
+    Post Table
+    """
+    postId = models.IntegerField(default=0)
+    publisher = models.ForeignKey(User, on_delete=models.CASCADE,
+                                  related_name="Publisher_Of_Post")
+    time = models.DateTimeField(auto_now=True)
+    markedUser = models.ManyToManyField(
+        User, related_name="User_Who_Marked_Post")
+    content = models.BinaryField()
```

Comment:
- Django gives each model default id field. Refer to https://docs.djangoproject.com/en/3.0/topics/db/models/#automatic-primary-key-fields
- A photo field and a description field is needed cause it is a photo-sharing application.
- We can add a `star` field to store stars the post received for a better recommendation.
- I suggest defining the `mark` relationship explicitly, which means creating a new table like `follow` relationship.
- What does the BiaryField `content` mean?

Reply:
+1.

1. `id` is auto populated, no need to define it manually. If you want this primary key to be named as other names, you can sure use `primary_key=True`, but you might need to use `AutoField` too. So I'd suggest we use the default one.
2. `related_name` is for reverse look up, that is, given a `User` object, to find all the posts he/she posted, `user.posts`, in your case, `user.Publisher_Of_Post` which resolves to an array of `Post`s, which is not correct.
And DO NOT use snake case with first char capitalized. I think the correct name should be `published_posts`.
3. `markedUser` does not follow snake_case naming convention.
4. `content` -> `description`, and it should be `TextField` I think, why do you need `BinaryField`? For Image, we don't actually save the binary data of the image in the db, instead, we save an URL path to the image in the db. The image data are actually saved somewhere else (media storage)


See also: https://docs.djangoproject.com/en/3.0/ref/models/fields/#django.db.models.Field.primary_key
https://docs.djangoproject.com/en/3.0/ref/models/fields/#autofield